content/articles/updating-spatial-database-via-flask-restful/index.md

```diff
@@ -0,0 +1,191 @@
+        return response, 204
+```
+
+We created a new class `CoordinateId`. ID was added to its name because we will retrieve coordinates using their ID. We added a new parameter `cordinate_id` to the get method because it will display the ID that is passed to the method. The ID is also passed to the search query in order to retrieve it from the database. If the ID is available it will be displayed and status code 200 is returned but if it's not available the default response message is displayed and the status code 204 is returned. The `coordinate_id` will be passed as a parameter to the URL also.
+```py
+api.add_resource(CoordinateId, '/api/coordinate/<int:coordinate_id>')
+```
+We can only pass an integer to the URL because we specify our variable type as an integer. The coordinate_id will be passed from the URL to the get method, which will be used to fetch the coordinate from our database.
+
+### Update Coordinate
+After retrieving a coordinate using its ID we can update it by sending a put request from Postman with the updated fields.
+```py
+ def put(self, coordinate_id):
+        response = {
+            "status": 204,
+            "message": "AOI is not available"
+        }
+        aoi_details = AoiCoordinate.query.filter_by(id=coordinate_id).first()
+        if aoi_details:
+            file = request.files['coordinate']
+            aoi = request.form.get('aoi')
+            read_file = file.read()
+            file_json = json.loads(read_file)
+            aoi_coordinate = file_json["features"][0]['geometry']
+
+            aoi_details.coordinate = aoi_coordinate
+            aoi_details.aoi = aoi
+            db.session.commit()
+
+            response['status'] = 200
+            response['message'] = "aoi updated successfully"
+            return response, 200
+        return response, 204
+```
+Just like the last get method, the Put method also accepts the coordinate_id that was passed from the URL, and it will use the same endpoint.
+
+### Delete
+This method is used for deleting the selected coordinate by sending a delete request from Postman. We will add the method to the CoordinateId class so that it will receive the coordinate_id.
+
+```py
+
+    def delete(self, coordinate_id):
+        response = {
+            "status": 204,
+            "message": "AOI is not available"
+        }
+        aoi = AoiCoordinate.query.filter_by(id=coordinate_id)
+        if aoi:
+            db.session.delete(aoi)
+            db.session.commit()
+            response['status'] = 200
+            response['message'] = 'Aoi deleted successfully'
+            return response, 200
+```
+
+If the coordinate is deleted successfully it will return the updated response code and status 200, but if not the default response will be returned.
+
+### Conclusion
+We have created an API that can be used for creating, reading, updating and deleting records from a database. It can be deployed for mobile, web and desktop apps.
+
```

Comment:
Add more content to your conclusion section. Provide more detail about what your article is about.

Reply:
I have fixed it.